eslint-rules/lib/rules/prop-value-shape-deprecation.js

```diff
@@ -0,0 +1,141 @@
+const _ = require('lodash');
+const utils = require('../utils');
+
+const MAP_SCHEMA = {
+  type: 'object',
+  properties: {
+    components: {
+      type: 'array',
+      items: {
+        type: 'string'
+      }
+    },
+    propNames: {
+      type: 'array',
+      items: {
+        type: 'string'
+      }
+    },
+    shape: {
+      type: 'array',
+      items: {
+        type: 'object',
+        required: ['prop', 'message', 'fix'],
+        properties: {
+          prop: {
+            type: 'string'
+          },
+          message: {
+            type: 'string'
+          },
+          fix: {
+            type: 'object',
+            additionalProperties: true
+          }
+        }
+      }
+    }
+  },
+  additionalProperties: true
+};
+
+module.exports = {
+  meta: {
+    docs: {
+      description: "Detect deprecation of prop's value shape",
+      category: 'Best Practices',
+      recommended: true
+    },
+    messages: {
+      uiLib: "This prop's value shape is deprecated."
+    },
+    fixable: 'code',
+    schema: [MAP_SCHEMA]
+  },
+  create(context) {
+    function testJSXAttributes(node) {
```

Comment:
This function is (way) too long, please divide into smaller functions.
I did not look at all the logic, I'll look closer after you'll refactor it.